share/eds/proofs_cache.go

```diff
@@ -317,15 +318,15 @@ func (c *proofsCache) getShare(rowIdx, colIdx int) ([]byte, error) {
 		if half.IsParity {
 			colIdx -= odsSize
 		}
-		return half.Shares[colIdx], nil
+		return half.Shares[colIdx].ToBytes(), nil
```

Comment:
previously, share was an alias to []byte so it worked fine. Since now it is `struct{[]byte}`, I decided to not break this function. I will come back to this function after all fixes to ensure that new share type will not significantly break the call stack